flink-connector-bigquery/src/test/java/com/google/cloud/flink/bigquery/table/restrictions/BigQueryPartitionTest.java

```diff
@@ -179,166 +181,311 @@ public void testWrongNumericPartition() {
-                        "event", BigQueryPartition.PartitionType.MONTH, StandardSQLTypeName.DATE);
-        Assertions.assertThat(
-                        BigQueryPartition.formatPartitionRestrictionBasedOnInfo(
-                                Optional.of(info), "event", "2010-10-10"))
-                .isEqualTo("event BETWEEN '2010-10' AND '2010-11'");
+    public void testCheckPartitionCompletedMonth() {
+        PartitionIdWithInfo partitionWithInfo =
+                new PartitionIdWithInfo(
+                        "202307",
+                        new TablePartitionInfo(
+                                "temporal",
+                                BigQueryPartition.PartitionType.MONTH,
+                                StandardSQLTypeName.DATE,
+                                // partition watermark is now
+                                Instant.now()));
+
+        PartitionIdWithInfoAndStatus partitionWithInfoAndStatus =
+                BigQueryPartition.checkPartitionCompleted(partitionWithInfo);
+
+        Assertions.assertThat(partitionWithInfoAndStatus.getStatus())
+                .isEqualTo(BigQueryPartition.PartitionStatus.COMPLETED);
     }
 
     @Test
-    public void testDateFormatYearPartitionRestriction() {
-        TablePartitionInfo info =
-                new TablePartitionInfo(
-                        "event", BigQueryPartition.PartitionType.YEAR, StandardSQLTypeName.DATE);
-        Assertions.assertThat(
-                        BigQueryPartition.formatPartitionRestrictionBasedOnInfo(
-                                Optional.of(info), "event", "2010-10-10"))
-                .isEqualTo("event BETWEEN '2010' AND '2011'");
+    public void testCheckPartitionCompletedYear() {
+        PartitionIdWithInfo partitionWithInfo =
+                new PartitionIdWithInfo(
+                        "2022",
+                        new TablePartitionInfo(
+                                "temporal",
+                                BigQueryPartition.PartitionType.YEAR,
+                                StandardSQLTypeName.DATE,
+                                // partition watermark is now
+                                Instant.now()));
+
+        PartitionIdWithInfoAndStatus partitionWithInfoAndStatus =
+                BigQueryPartition.checkPartitionCompleted(partitionWithInfo);
+
+        Assertions.assertThat(partitionWithInfoAndStatus.getStatus())
+                .isEqualTo(BigQueryPartition.PartitionStatus.COMPLETED);
     }
 
-    @Test(expected = IllegalArgumentException.class)
-    public void testDateFormatHourPartitionRestriction() {
-        TablePartitionInfo info =
-                new TablePartitionInfo(
-                        "event", BigQueryPartition.PartitionType.HOUR, StandardSQLTypeName.DATE);
-        BigQueryPartition.formatPartitionRestrictionBasedOnInfo(
-                Optional.of(info), "event", "2010-10-10");
+    @Test
+    public void testCheckPartitionNotCompletedYear() {
```

Comment:
We should add edge case tests for leap year.

Reply:
Done